src/app/model/Pawn.ts

```diff
@@ -0,0 +1,70 @@
+import { Board } from './Board';
+import { BOARDSIZE } from './Constances';
+import { Piece } from './Piece';
+import { PieceMovement } from './PieceMovement';
+import { Column, columns, Row, Side, Square } from './Types';
+
+export class Pawn extends Piece implements PieceMovement {
+  private nextRowDifference: number = 1;
+
+  constructor(id: string, side: Side) {
+    super(id, side);
+    if (this.side !== 'WHITE') this.nextRowDifference = -1;
```

Comment:
Aaa widze, tutaj zmieniasz. Co powiesz na np. 
```
const NEXT_ROW_DIFFERENCE = {
WHITE: 1,
BLACK: -1
}
```

potem jak masz side to dostaniesz sie do tego:
`NEXT_ROW_DIFFERENCE[side]` i mozesz pozbyc sie zapisu z ifem. Im mniej ifów tym lepiej :) Powstają nawet całe kursy anti-if.

Reply:
Brzmi świetnie!

Reply:
Czy to wrzucić w konstruktor, w klasę czy poza klasę?

Reply:
Poza klasę chyba będzie sensownie. Bo zawsze jest to samo - niezależnie od instancni.